src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs

```diff
@@ -72,8 +72,12 @@ protected void WaitForReceived(int numberOfMessages = 1)
     {
         long aMessageCount = aClient.WaitCount += numberOfMessages;
         long bMessageCount = bClient.WaitCount += numberOfMessages;
-        while (!_externalProcessRunner.ErrorTriggerEncountered && aClient.MessageCount < aMessageCount ||
-               bClient.MessageCount < bMessageCount)
+        
+        var noErrorEncountered = !_externalProcessRunner.ErrorTriggerEncountered;
+        var aClientMessageMissing = aClient.MessageCount < aMessageCount;
+        var bClientMessageMissing = bClient.MessageCount < bMessageCount;
+        
+        while (noErrorEncountered && (aClientMessageMissing || bClientMessageMissing))
```

Comment:
But the values of the `noErrorEncountered` and two other variables are not recomputed after each `Thread.Sleep(...)`, so how can this while-loop exit if it runs its body at least once?